truenas_installer/installer_menu.py

```diff
@@ -0,0 +1,165 @@
+import asyncio
+import os
+import sys
+
+import humanfriendly
+
+from .dialog import dialog_checklist, dialog_menu, dialog_msgbox, dialog_password, dialog_yesno
+from .disks import list_disks
+from .install import InstallError, install
+from .serial import serial_sql
+from .swap import is_swap_safe
+
+
+class InstallerMenu:
+    def __init__(self, installer):
+        self.installer = installer
+
+    async def run(self):
+        await self._main_menu()
+
+    async def _main_menu(self):
+        await dialog_menu(
+            f"TrueNAS {self.installer.version} Console Setup",
+            {
+                "Install/Upgrade": self._install_upgrade,
+                "Shell": self._shell,
+                "Reboot System": self._reboot,
+                "Shutdown System": self._shutdown,
+            }
+        )
+
+    async def _install_upgrade(self):
+        while True:
+            await self._install_upgrade_internal()
+            await self._main_menu()
+
+    async def _install_upgrade_internal(self):
+        disks = await list_disks()
+
+        if not disks:
+            await dialog_msgbox("Choose destination media", "No drives available")
+            return False
+
+        while True:
+            destination_disks = await dialog_checklist(
+                "Choose Destination Media",
+                (
+                    "Install TrueNAS to a drive. Multiple drives can be selected to provide redundancy. "
+                    "Chosen drives are not available for use in the TrueNAS UI. Arrow keys highlight options, "
+                    "spacebar selects."
+                ),
+                {
+                    disk.name: " ".join([
+                        disk.model[:15].ljust(15, " "),
+                        disk.label[:15].ljust(15, " "),
+                        "--",
+                        humanfriendly.format_size(disk.size, binary=True)
+                    ])
+                    for disk in disks
+                }
+            )
+
+            if destination_disks is None:
```

Comment:
When will destination_disks be `None` as opposed to an empty list? Maybe add a comment about why this would happen.